crates/router/src/workflows/outgoing_webhook_retry.rs

```diff
@@ -34,202 +38,379 @@ use crate::{
+#[cfg(feature = "v1")]
+#[instrument(skip_all)]
+pub async fn start_outgoing_webhook_workflow(
+    state: &SessionState,
+    process_tracker: &storage::ProcessTracker,
+) -> Result<(), errors::ProcessTrackerError> {
+    match process_tracker.name.as_deref() {
+        Some(OUTGOING_WEBHOOK_BULK_TASK) => {
+            Box::pin(trigger_outgoing_webhook_bulk_workflow(
+                state,
+                process_tracker,
+            ))
+            .await
+        }
+        Some(OUTGOING_WEBHOOK_RETRY_TASK) => {
+            Box::pin(trigger_outgoing_webhook_retry_workflow(
+                state,
+                process_tracker,
+            ))
+            .await
+        }
+        Some(unexpected_job) => Err(errors::ApiErrorResponse::WebhookProcessingFailure)
+            .attach_printable(format!(
+                "Unexpected task name encountered: `{}`",
+                unexpected_job
+            ))?,
+        _ => Err(errors::ProcessTrackerError::JobNotFound),
+    }
+}
+
+#[instrument(skip_all)]
+#[cfg(feature = "v1")]
+pub async fn trigger_outgoing_webhook_bulk_workflow(
+    state: &SessionState,
+    process: &storage::ProcessTracker,
+) -> Result<(), errors::ProcessTrackerError> {
+    let tracking_data: OutgoingWebhookTrackingData = process
+        .tracking_data
+        .clone()
+        .parse_value("OutgoingWebhookTrackingData")?;
+    let merchant_id = tracking_data.merchant_id.clone();
+    let profile_id = tracking_data.business_profile_id.clone();
+    let primary_object_id = tracking_data.primary_object_id.clone();
+    let primary_object_created_at = tracking_data.primary_object_created_at;
+    let event_type = tracking_data.event_type;
+    let event_class = tracking_data.event_class;
+    let primary_object_type = tracking_data.primary_object_type;
+    let db = &*state.store;
+    let key_manager_state = &state.into();
+
+    let merchant_key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            key_manager_state,
+            &merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await?;
+    let merchant_account = db
+        .find_merchant_account_by_merchant_id(key_manager_state, &merchant_id, &merchant_key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    let business_profile = utils::validate_and_get_business_profile(
+        db,
+        key_manager_state,
+        &merchant_key_store,
+        Some(&profile_id),
+        &merchant_id,
+    )
+    .await?
+    .ok_or(errors::ApiErrorResponse::ProfileNotFound {
+        id: profile_id.get_string_repr().to_string(),
+    })?;
+
+    let cloned_state = state.clone();
+    let (content, _) = Box::pin(get_outgoing_webhook_content_and_event_type(
+        state.clone(),
+        state.get_req_state(),
+        merchant_account.clone(),
+        merchant_key_store.clone(),
+        &tracking_data,
+    ))
+    .await?;
```

Comment:
`get_outgoing_webhook_content_and_event_type()` isn't meant to be used as the primary way of obtaining the outgoing webhook content: it was added for backward compatibility reasons.

Another reason I would prefer to completely avoid it is that it has the caveat that the resource could have transitioned to a completely different status (as compared to what we're notifying about) by the time the workflow would be executed.

We previously had checks for it:
https://github.com/juspay/hyperswitch/blob/0a06ad91252d76d0c937344d8c76c9a2bbbc84d7/crates/router/src/workflows/outgoing_webhook_retry.rs#L175-L228